stdlib/buffer.mli

```diff
@@ -158,6 +158,12 @@ val truncate : t -> int -> unit
   Raise [Invalid_argument] if [len < 0] or [len > length b].
   @since 4.05.0 *)
 
+val drop : t -> int -> unit
```

Comment:
Given how buffer are used in practice, I find that to be a confusing name, if I read `Buffer.drop` I might expect it to drop bytes at the end. I would rather suggest `Buffer.drop_{left,right}`.

Reply:
How about `advance`?

Reply:
Again I don't find `advanc`ing an append right data structure to be very enlighting. I would prefer to have both `Buffer.drop_left` and `Buffer.drop_right`.

Reply:
I'm happy to use whatever has the consensus. `drop_right` would have to be an alias to `truncate`, which already exists. I don't really like aliases personally.

Reply:
No `truncate len` sets the length to `len`. `drop_{left,right} n` remove `n` bytes.

Reply:
```
let drop_right b n =
  truncate b (length b - n)
```
Looks pretty much like an alias to me..

Reply:
Depends on your definition of alias... for me aliases can be substituted one for each other. That is not the case here.